src/replay/replay_recorder.cpp

```diff
@@ -116,13 +129,125 @@ void ReplayRecorder::update(int ticks)
 #ifdef DEBUG
         m_count++;
 #endif
-        if (time - m_last_saved_time[i] < stk_config->m_replay_dt)
+        // If one of the tracked kart data has significantly changed
+        // for the kart, update sooner than the usual dt
+        bool force_update = false;
+
+        // Don't save directly the enum value, because any change
+        // to it would break the reading of old replays
+        int attachment = enumToCode(kart->getAttachment()->getType());
+        int powerup_type = enumToCode(kart->getPowerup()->getType());
+        int special_value = 0;
+
+        // In egg hunt mode, use store the number of eggs found so far
+        // This assumes that egg hunt mode is only available in single-player
+        // TODO : when adding battle mode support, store the number of lives here
+        if (race_manager->isEggHuntMode())
+        {
+            //TODO : figure out why the compiler don't like calling numberOfEggsFound
```

Comment:
Well that should be figured out and the TODO removed before this can be merged :)